packages/lib/src/vis/hooks.ts

```diff
@@ -111,59 +111,36 @@ export function useCssColors(
   return [colors, refCallback];
 }
 
-export function useGeometry<
-  AttributeNames extends string,
-  Params extends object,
->(
-  Ctor: new (len: number) => H5WebGeometry<AttributeNames, Params>,
-  dataLength: number,
-  params: Params,
+export function useUpdateGeometry(
+  geometry: H5WebGeometry & BufferGeometry,
   config: {
     skipUpdates?: boolean; // set to `true` when R3F object is hidden
     isInteractive?: boolean; // set to `true` to keep bounding sphere up to date for raycaster
   } = {},
-): H5WebGeometry<AttributeNames, Params> {
+): void {
   const { skipUpdates = false, isInteractive = false } = config;
-
-  const geometry = useMemo(() => new Ctor(dataLength), [Ctor, dataLength]);
   const invalidate = useThree((state) => state.invalidate);
 
-  useLayoutEffect(
-    () => {
-      if (skipUpdates) {
-        return;
-      }
-
-      geometry.prepare(params);
-
-      for (let i = 0; i < dataLength; i += 1) {
-        geometry.update(i);
-      }
-
-      if (isInteractive) {
-        geometry.computeBoundingSphere(); // https://github.com/mrdoob/three.js/issues/1170#issuecomment-3617180
-      }
-
-      Object.values<BufferAttribute>(geometry.attributes).forEach((attr) => {
-        attr.needsUpdate = true; // eslint-disable-line no-param-reassign
-      });
-
-      if (geometry.index) {
-        geometry.index.needsUpdate = true;
-      }
-
-      invalidate();
-    },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [
-      geometry,
-      dataLength,
-      ...Object.values(params), // eslint-disable-line react-hooks/exhaustive-deps
-      skipUpdates,
-      isInteractive,
-      invalidate,
-    ],
-  );
+  useLayoutEffect(() => {
+    if (skipUpdates) {
+      return;
+    }
+
+    geometry.update();
+
+    if (isInteractive) {
+      geometry.computeBoundingBox();
```

Comment:
`computeBoundingSphere` calls this internally if needed, but I think it's best to be explicit.